File: workspace/catkin_ws/src/simple_comm_cpp/src/listener.cpp
#include "ros/ros.h"
#include "std_msgs/String.h"

// Callback function that will be called when a message is received
void chatterCallback(const std_msgs::String::ConstPtr& msg)
{
    // Log the received message to the console
    ROS_INFO("I heard: [%s]", msg->data.c_str());
}

int main(int argc, char **argv)
{
    // Initialize the ROS node
    ros::init(argc, argv, "listener");

    // Create a handle for the node
    ros::NodeHandle n;

    // Create a subscriber that subscribes to the "chatter" topic
    ros::Subscriber sub = n.subscribe("chatter", 1000, chatterCallback);

    // Keep the node running to listen for incoming messages
    ros::spin();

    return 0;
}

File: workspace/catkin_ws/src/python_pkg/src/listener.py
#!/usr/bin/env python

import rospy
from std_msgs.msg import String

class Listener:
    def __init__(self):
        rospy.init_node('listener', anonymous=True)
        rospy.Subscriber('chatter', String, callback)
        
    def callback(data):
        rospy.loginfo("I heard: %s", data.data)

    def listen():
        rospy.spin()

if __name__ == '__main__':
    listener = Listener()
    listener.listen()
